### PR TITLE
NodeMaterial: updates and revision

### PR DESCRIPTION
**Pull request by sunag**
**Description**

## NodeMaterial & WebGPU

- [x] `physicallyCorrectLights` is default in `WebGPURenderer`  
- [x] Split all `BRDFs` in individual files
- [x] `ToneMappingNode` support
   - use `renderer.toneMappingNode = new Nodes.ToneMappingNode( ... )`
- [x] add `FogNode.mix()`
- [x] add `NodeMaterial.generateOutput()` and revision to add `IBL` soon
- [x] add `WebGPUNodeBuilder.getSamplerBias()`
- [x] `VarNode.op( ... )` support
- [x] add `ReflectedLightNode`
- [x] update examples

## ShaderNode

- [x] `ShaderNode( ... )` was deprecated, use `ShaderNode.call( .. )` instead of to call a function
- [x] `addTo( varNode, ... )` was deprecated, use `varNode.add( ...  )` instead of
- [x] add `context`, `reflectedLight`

## NodePlayground

- [x] Update `WebGPU` and `WebGL` examples

*This contribution is funded by [Google via Igalia](https://igalia.com)*

### CODE REVIEW

File: examples/jsm/nodes/functions/PhysicalLightingModel.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	const dotNL = saturate( dot( transformedNormalView, lightDirection ) );
+	const irradiance = mul( dotNL, lightColor );
+
+	reflectedLight.directSpecular.add( mul( irradiance, BRDF_GGX.call( { lightDirection, f0: specularColor, f90: 1, roughness } ) ) );
+
+	reflectedLight.directDiffuse.add( mul( irradiance, BRDF_Lambert.call( { diffuseColor: diffuseColor.rgb } ) ) );
+
+} );
+
+const PhysicalLightingModel = new ShaderNode( ( inputs/*, builder*/ ) => {
```
Comment (LeviPesin) [inline examples/jsm/nodes/functions/PhysicalLightingModel.js:22]: Can it be just `const PhysicalLightingModel = RE_Direct_Physical`?

Comment (sunag) [reply examples/jsm/nodes/functions/PhysicalLightingModel.js:22]: At this moment it make more sense but I will add others functions like `RE_IndirectSpecular_Physical` soon...

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
> - use `scene.toneMappingNode = new Nodes.ToneMappingNode( ... )`

Hmm, do you think we should move `renderer.toneMapping` to `scene.toneMapping`? 🤔

**Comment by sunag** [comment]
> Hmm, do you think we should move renderer.toneMapping to scene.toneMapping? 🤔

No, no! I made a mistake in the description only. 
https://github.com/sunag/three.js/blob/bbb7841dcfcb53e1657de737c18b3ff8c6fd649b/examples/webgpu_nodes_playground.html#L111

**Pull request merged**